src/inheritAttributes.js

```diff
@@ -287,6 +287,45 @@ export const parseCaptionServiceMetadata = (service) => {
   }
 };
 
+/**
+ * A map callback that will parse all event stream data for a collection of periods
+ * DASH ISO_IEC_23009 5.10.2.2
+ * https://dashif-documents.azurewebsites.net/Events/master/event.html#mpd-event-timing
+ *
+ * @param {PeriodInformation} period object containing necessary period information
+ * @return a collection of parsed eventstream event objects
+ */
+export const toEventStream = (period) => {
+  // get and flatten all EventStreams tags and parse attributes and children
+  return flatten(findChildren(period.node, 'EventStream').map((eventStream) => {
+    const eventStreamAttributes = parseAttributes(eventStream);
+    const schemeIdUri = eventStreamAttributes.schemeIdUri;
+    // schemeIdUri is mandatory for EventStream tags
+
+    if (!schemeIdUri) {
+      return;
+    }
+
+    // find all Events per EventStream tag and map to return objects
+    return findChildren(eventStream, 'Event').map((event) => {
+      const eventAttributes = parseAttributes(event);
+      const presentationTime = eventAttributes.presentationTime || 0;
+      const timescale = eventStreamAttributes.timescale || 1;
+      const duration = eventAttributes.duration || 0;
+      const start = (presentationTime / timescale) + period.attributes.start;
+
+      return {
```

Comment:
According to the spec `Event` elements also have a `@contentEncoding` attribute. It's optional, but may be worth including.

Reply:
Good call. Adding this to the Event object.